cmd/snap-confine/snap-confine-invocation.c

```diff
@@ -83,19 +83,23 @@ void sc_cleanup_invocation(sc_invocation *inv) {
         sc_cleanup_string(&inv->orig_base_snap_name);
         sc_cleanup_string(&inv->security_tag);
         sc_cleanup_string(&inv->executable);
+        sc_cleanup_string(&inv->rootfs_dir);
     }
 }
 
-void sc_maybe_pick_alt_base_snap(sc_invocation *inv) {
-    /* As a special fallback, allow the base snap to degrade from "core" to
-     * "ubuntu-core". This is needed for the migration tests. */
+void sc_check_init_rootfs_dir(sc_invocation *inv) {
     char mount_point[PATH_MAX] = {0};
     sc_must_snprintf(mount_point, sizeof mount_point, "%s/%s/current/", SNAP_MOUNT_DIR, inv->base_snap_name);
 
     if (access(mount_point, F_OK) == 0) {
+        inv->rootfs_dir = sc_strdup(mount_point);
```

Comment:
I would rather initialise the root directory in `sc_init_invocation` and only change it here if necessary.